### PR TITLE
k8s: update eni-max-pods

### PR DESCRIPTION
**Pull request by samjo-nyang**
Add m6i instance types and others: fetched from https://github.com/awslabs/amazon-eks-ami/pull/749/files

**Issue number:** #1720

**Description of changes:** Update eni-max-pods (add m6i instances)

**Testing done:** (none)

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by zmrow — APPROVED** [review]
Thanks for doing this!

I pulled down the file from the [associated PR](https://github.com/awslabs/amazon-eks-ami/pull/735/files) on the EKS AMI repo and you've covered the diff here.  :+1: 

:1st_place_medal:

**Comment by zmrow** [comment]
Question for others - I don't foresee any issues with this, but perhaps we should hold off merging until the EKS PR gets approved and merged?  Just in case there are any changes made over there?

**Comment by samjo-nyang** [comment]
@zmrow Hi, I found that https://github.com/awslabs/amazon-eks-ami/pull/749/files pr is merged, instead of previous one. I just force-pushed it.

**Review by jpculp — APPROVED** [review]
🚀

**Comment by zmrow** [comment]
Thanks again!

**Review by zmrow — APPROVED** [review]

**Pull request merged**
